Bukkit/src/main/java/com/plotsquared/bukkit/listener/EntityEventListener.java

```diff
@@ -243,6 +239,32 @@ public void onEntityFall(EntityChangeBlockEvent event) {
         }
     }
 
+    @EventHandler(ignoreCancelled = true, priority = EventPriority.HIGHEST)
+    public void onWeavingEffect(EntityChangeBlockEvent event) {
+        if (event.getTo() != Material.COBWEB) {
+            return;
+        }
+        Block block = event.getBlock();
+        World world = block.getWorld();
+        String worldName = world.getName();
+        if (!this.plotAreaManager.hasPlotArea(worldName)) {
+            return;
+        }
+        Location location = BukkitUtil.adapt(block.getLocation());
+        PlotArea area = location.getPlotArea();
+        if (area == null) {
+            return;
+        }
+        Plot plot = area.getOwnedPlotAbs(location);
+
+        if (plot == null || !plot.getFlag(WeavingDeathPlace.class)) {
```

Comment:
In case of `null`-plot, handle road flags (`PlotFlagUtil.isAreaRoadFlagsAndFlagEquals`)

Reply:
Now, I use the same checks as `onDamage(EntityDamageEvent event)` below with separate handling of the road flags.